src/sentry_plugins/bitbucket/client.py

```diff
@@ -49,20 +82,99 @@ def create_issue(self, repo, data):
         }
         return self.request(
             'POST',
+            '1.0',
             '/repositories/%s/issues' % (repo,),
             data=data
         )
 
     def search_issues(self, repo, query):
         return self.request(
             'GET',
+            '1.0',
             '/repositories/%s/issues' % (repo,),
             params={'search': query},
         )
 
     def create_comment(self, repo, issue_id, data):
         return self.request(
             'POST',
+            '1.0',
             '/repositories/%s/issues/%s/comments' % (repo, issue_id),
             data=data,
         )
+
+    # copied from github
+
+    def get_repo(self, repo):
+        return self.request(
+            'GET',
+            '2.0',
+            '/repositories/{}'.format(repo),
+        )
+
+    def create_hook(self, repo, data):
+        return self.request(
+            'POST',
+            '2.0',
+            '/repositories/{}/hooks'.format(
+                repo,
+            ),
+            data=data,
+        )
+
+    def delete_hook(self, repo, id):
+        return self.request(
+            'DELETE',
+            '2.0',
+            '/repositories/{}/hooks/{}'.format(
+                repo,
+                id,
+            ),
+        )
+
+    def get_commit_filechanges(self, repo, sha):
+        # return api request that fetches last ~30 commits
+        # see https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D/commits/%7Brevision%7D
+        # using sha as parameter
+        patch_file = self.request(
+            'GET',
+            '2.0',
+            '/repositories/{}/patch/{}'.format(
+                repo,
+                sha,
+            )
+        )
+        patch = PatchSet(patch_file, encoding='utf-8')
+        return patch
+
+    def get_last_commits(self, repo, end_sha):
+        # return api request that fetches last ~30 commits
+        # see https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D/commits/%7Brevision%7D
+        # using end_sha as parameter
+        return self.request(
+            'GET',
+            '2.0',
+            '/repositories/{}/commits/{}'.format(
+                repo,
+                end_sha,
+            )
+        )
+
+    def compare_commits(self, repo, start_sha, end_sha):
+        # where start sha is oldest and end is most recent
+        # see https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D/commits/%7Brevision%7D
+        data = self.request(
+            'GET',
+            '2.0',
+            '/repositories/{}/commits/{}'.format(
+                repo,
+                end_sha
+            )
+        )
+        commits = []
+        for commit in data['values']:
+            # TODO(maxbittker) fetch extra pages when this is paginated
```

Comment:
if/when you need to paginate, we should have a maximum number of pages we're willing to paginate through since the start commit is user data and could possibly be invalid